markdown/events/2023/bytesize_mag.md

```diff
@@ -21,3 +24,89 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[10:32](https://www.youtube.com/watch?v=IiorfDHeoLo&t=632)
+Another topic I would like to shortly mention is the resource requirements, because this came up quite often in the Slack channel, and it's also somehow difficult to estimate in advance, because it really differs depending on the input data. The main requirements are both for memory and time, coming from the assembly step. As I mentioned already it really differs for different input data sets and I collected some numbers just to give you a rough idea for different pipeline ones that were run by Daniel Straub on our compute cluster. For one rather small sample, which was a cultural sample, both MegaHIT and SPAdes required less than 25 gigabytes and were finished in a couple of hours. However, for a larger river sample data set, MegaHIT took already more than 100 gigabytes of RAM, and it took more than one day to finish, and SPAdes even took more than 900 gigabytes of memory, and it required more than nine days. There was another very large data set containing 15 soil samples for which also a core assembly was performed and for this MegaHIT required one terabyte and more than 17 days, and SPAdes could not even be run because it would have required more than two terabytes of memory.
+
+[11:55](https://www.youtube.com/watch?v=IiorfDHeoLo&t=715)
+This just shows that even for smaller data sets, you cannot run this on your laptop. In general, one can say that it depends on the sequencing depths, the number of samples, the complexity of the analyte metagenome, and also on the applied tool and setting. For this it might be worth noting that, that both assembly tools are run by default but MegaHIT requires much fewer resources than SPAdes, and if you do not want to compute a hybrid assembly it might make sense to consider the `--skip_spades` parameter. Additionally, the core assembly also increases the required resources because it pools samples. At least for one individual task, the required memory and time is much higher. This is something important to keep in mind, because also if you want to run it on larger data sets, you might want to provide a custom config file in order to adjust the resources required for your particular data set.
```

Comment:
```suggestion
This just shows that even for smaller data sets, you cannot run this on your laptop. In general, one can say that it depends on the sequencing depths, the number of samples, the complexity of the underlying metagenome, and also on the applied tool and setting. For this it might be worth noting that both assembly tools are run by default but MegaHIT requires much fewer resources than SPAdes, and if you do not want to compute a hybrid assembly it might make sense to consider the `--skip_spades` parameter. Additionally, the co-assembly also increases the required resources because it pools samples. At least for one individual task, the required memory and time is much higher. This is something important to keep in mind, because also if you want to run it on larger data sets, you might want to provide a custom config file in order to adjust the resources required for your particular data set.
```